### PR TITLE
Make Contravariant a subclass of InvariantFunctor

### LINKED ISSUE #1859: Contravariant should be a subclass of InvariantFunctor
`Functor` already is. Just do the same thing.

### PR DESCRIPTION
**Pull request by earldouglas**
Fixes #1859

### CODE REVIEW

File: base/shared/src/main/scala/scalaz/ct/phantom.scala [hunk 1]
```diff
@@ -6,7 +6,10 @@ import scalaz.data.Const
 import scala.language.experimental.macros
 
 trait PhantomClass[F[_]] extends FunctorClass[F] with ContravariantClass[F] {
+
   def pmap[A, B](ma: F[A]): F[B]
+
+  override def imap[A, B](ma: F[A])(f: A => B)(g: B => A): F[B] = pmap(ma)
 }
```
Comment (hrhino) [inline base/shared/src/main/scala/scalaz/ct/phantom.scala]: arguably you can also add overrides for `map` and `contramap` but that doesn't need to be done now.

Comment (earldouglas) [reply base/shared/src/main/scala/scalaz/ct/phantom.scala]: Would that make `DeriveMapContramap` redundant?

Comment (hrhino) [reply base/shared/src/main/scala/scalaz/ct/phantom.scala]: ah, yeah, we're doing it there. I'm working on removing the `Derive*` traits, so don't worry.

**Comment by hrhino** [comment]
Thanks, @earldouglas!!

**Pull request merged**